.github/workflows/deploy.yml

```diff
@@ -85,9 +85,62 @@ jobs:
           if-no-files-found: error
           retention-days: 20
 
-  deploy:
-    name: Build and push docker image
+  integration-tests:
     needs: build
+    name: Run integration tests suite
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout Source code
+        uses: actions/checkout@v2
+
+      - name: Download artifact with aleph-node binary
+        uses: actions/download-artifact@v2
+        with:
+          name: aleph-node
+
+      - name: Build docker image
+        id: build-image
+        run: |
+          mkdir -p target/release/
+          # copy artifact
+          cp aleph-node target/release/aleph-node
+          # Build and tag a docker container
+          docker build --tag aleph-node:latest -f ./docker/Dockerfile .
+
+      - name: Run consensus party
+        run: |
+          docker-compose -f docker/docker-compose.yml up &
+
+      - name: Verify nodes respond
+        run: |
+          validators=( damian tomasz zbyszko hansu )
+
+          for validator in "${validators[@]}"; do
+            docker run --network container:$validator appropriate/curl:latest --retry 3 --retry-delay 10 -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "rpc_methods"}' http://127.0.0.1:9933
```

Comment:
Could you break this line with '\'? Also, a small comment what this call exactly do would be helpful.

Reply:
Done